src/main/kotlin/apply/application/ApplicantService.kt

```diff
@@ -20,17 +20,15 @@ class ApplicantService(
         return applicantRepository.findByEmail(email) ?: throw IllegalArgumentException("지원자가 존재하지 않습니다. email: $email")
     }
 
-    fun findAllByRecruitmentIdAndKeyword(recruitmentId: Long, keyword: String): List<ApplicantAndFormResponse> {
-        return applicationFormRepository
-            .findByRecruitmentIdAndSubmittedTrue(recruitmentId)
-            .associateBy { it.applicantId }
-            .match { applicantRepository.findAllByKeyword(keyword) }
-    }
-
-    fun findAllByRecruitmentIdAndSubmittedTrue(recruitmentId: Long): List<ApplicantAndFormResponse> {
-        return applicationFormRepository
+    fun findAllByRecruitmentIdAndSubmittedTrueAndKeyword(recruitmentId: Long, keyword: String?): List<ApplicantAndFormResponse> {
+        val applicationFormMap = applicationFormRepository
             .findByRecruitmentIdAndSubmittedTrue(recruitmentId)
             .associateBy { it.applicantId }
+        if (keyword != null) {
+            return applicationFormMap
+                .match { applicantRepository.findAllByKeyword(keyword) }
+        }
+        return applicationFormMap
             .run { match { applicantRepository.findAllById(keys) } }
     }
```

Comment:
```suggestion
    fun findAllByRecruitmentIdAndSubmittedTrueAndKeyword(recruitmentId: Long, keyword: String?): List<ApplicantAndFormResponse> {
        val applicationFormMap = applicationFormRepository
            .findByRecruitmentIdAndSubmittedTrue(recruitmentId)
            .associateBy { it.applicantId }
        return if (keyword != null) {
            applicationFormMap.match { applicantRepository.findAllByKeyword(keyword) }
        } else {
            applicationFormMap.run { match { applicantRepository.findAllById(keys) } }
        }
```
keyword를 nullable하게 받는 방식 너무 좋네요. 👍  

참고로, 코틀린의 if문은 expression이기 때문에 값을 리턴합니다. 그래서 이런 식으로도 쓸 수 있어요. 아니면 `?.`와 `?:`를 활용해서도 처리할 수 있겠네요. 

----
그리고 조금 더 리팩토링을 진행해봐도 좋을 거 같아요! 안그래도 `match()`메서드가 계속 눈에 밟혔거든요. 중복을 해결하기 위해 만든 메서드였는데 이제 중복도 없어졌으니 가독성을 올려봐도 될 거 같아요. `match()` 메서드를 제거하고 `findAllByRecruitmentIdAndSubmittedTrueAndKeyword()` 메서드만 남겨도 될 것 같지 않나요?

Reply:
말씀대로 `findAllByRecruitmentIdAndSubmittedTrueAndKeyword()`을 좀 더 개선해보았습니다!!!!
제안해주신 분기 처리를 지원자리스트 받아오는데 적용하고 `match()`를 제거해보았습니다
아직 코틀린 문법이 익숙치 않은데 좋은 제안 덕분에 많이 배워갑니다 
부족한 부분있으면 말씀부탁드립니다 감사합니다 🙇‍♂️